slothy/core/core.py

```diff
@@ -855,7 +855,7 @@ def selftest(self, log):
                 addr = getattr(t.inst, "addr", None)
                 if addr is None:
                     continue
-                addresses.add(addr)
+                addresses = addresses.union(tree.find_all_predecessors_input_registers(t, addr))
```

Comment:
Blocking: This uses the final `reg_state` -- it should use the `reg_state` at the time of `t`.